istioctl/pkg/writer/envoy/configdump/xds.go

```diff
@@ -0,0 +1,43 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// nolint: lll
+//
+//go:generate sh -c "echo '//go:build !agent' > filter_types.gen.go"
+//go:generate sh -c "echo '// +build !agent\n' >> filter_types.gen.go"
+//go:generate sh -c "echo '// Copyright Istio Authors' >> filter_types.gen.go"
+//go:generate sh -c "echo '//' >> filter_types.gen.go"
+//go:generate sh -c "echo '// Licensed under the Apache License, Version 2.0 (the \"License\");' >> filter_types.gen.go"
+//go:generate sh -c "echo '// you may not use this file except in compliance with the License.' >> filter_types.gen.go"
+//go:generate sh -c "echo '// You may obtain a copy of the License at' >> filter_types.gen.go"
+//go:generate sh -c "echo '//' >> filter_types.gen.go"
+//go:generate sh -c "echo '//     http://www.apache.org/licenses/LICENSE-2.0' >> filter_types.gen.go"
+//go:generate sh -c "echo '//' >> filter_types.gen.go"
+//go:generate sh -c "echo '// Unless required by applicable law or agreed to in writing, software' >> filter_types.gen.go"
+//go:generate sh -c "echo '// distributed under the License is distributed on an \"AS IS\" BASIS,' >> filter_types.gen.go"
+//go:generate sh -c "echo '// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.' >> filter_types.gen.go"
+//go:generate sh -c "echo '// See the License for the specific language governing permissions and' >> filter_types.gen.go"
+//go:generate sh -c "echo '// limitations under the License.\n' >> filter_types.gen.go"
+//go:generate sh -c "echo '//  GENERATED FILE -- DO NOT EDIT\n' >> filter_types.gen.go"
+//go:generate sh -c "echo 'package configdump\n\nimport (' >> filter_types.gen.go"
+//go:generate sh -c "go list github.com/envoyproxy/go-control-plane/... | grep 'v[2-9]' | grep -v /pkg/ | xargs -I{} echo '\t_ \"{}\"' >> filter_types.gen.go"
+//go:generate sh -c "echo '\n\t// Istio-specific Envoy filters' >> filter_types.gen.go"
+//go:generate sh -c "go list istio.io/api/envoy/config/filter/... | grep 'v[2-9]' | xargs -I{} echo '\t_ \"{}\"' >> filter_types.gen.go"
+//go:generate sh -c "echo ')' >> filter_types.gen.go"
+package configdump
+
+// Import all Envoy filter types so they are registered and deserialization does not fail
```

Comment:
why do we need this? can we just import `https://github.com/istio/istio/pull/42365#`?

Reply:
I import this for some proto unmarshal issue, let me find out if these still needed.

Reply:
Sorry I pasted wrong thing. I mean just import pkg/config/xds/filter_types.gen.go, don't duplicate it

Reply:
yes, that's the right way.

Reply:
we don't need this anymore for using `protojson`